monkey/common/event_queue/i_event_queue.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractstaticmethod
+from typing import Any, Callable, Sequence
+
+from common.events import AbstractEvent
+
+
+class IEventQueue(ABC):
+    """
+    Manages subscription and publishing of events
+    """
+
+    @abstractstaticmethod
+    def subscribe_all(subscriber: Callable[..., Any]):
+        """
+        Subscribes a subscriber to all events
+
+        :param subscriber: Callable that should subscribe to events
+        """
+
+        pass
+
+    @abstractstaticmethod
+    def subscribe_types(types: Sequence[AbstractEvent], subscriber: Callable[..., Any]):
+        """
+        Subscribes a subscriber to all specifed event types
+
+        :param types: Event types to which the subscriber should subscribe
+        :param subscriber: Callable that should subscribe to events
+        """
+
+        pass
+
+    @abstractstaticmethod
+    def subscribe_tags(tags: Sequence[str], subscriber: Callable[..., Any]):
```

Comment:
I'm not totally sure whether or not the most common case will be to subscribe to a single tag or multiple tags. My gut says "single tag", but I can't justify that with anything objective like I can for `subscribe_types()`.

I think I'd prefer we modify the interface to subscribe to a single tag for the following reasons:
1. It makes the interface and components simpler to use and test
2. Subscription will probably only happen in one place. The likelihood of introducing duplicate code that iterates tags and calls `subscribe_tags()` repeatedly is low.